lib/improver/blending/blend_across_adjacent_points.py

```diff
@@ -132,20 +132,20 @@ def _find_central_point(self, cube):
         """
         # Convert central point into the units of the cube, so that a
         # central point can be extracted.
-        self.central_point = (
+        central_point = (
             Unit(self.parameter_units).convert(
                 self.central_point, cube.coord(self.coord).units))
         constr = iris.Constraint(
-            coord_values={self.coord: self.central_point})
+            coord_values={self.coord: central_point})
         central_point_cube = cube.extract(constr)
         if central_point_cube is None:
             if self.parameter_units is None:
                 parameter_units = central_point_cube.coord(self.coord).units
             else:
                 parameter_units = self.parameter_units
-            msg = ("The central point of {} in units of {} not available "
+            msg = ("The central point {} in units of {} not available "
                    "within input cube coordinate points: {}.".format(
-                       self.central_point, parameter_units,
+                       central_point, parameter_units,
```

Comment:
This function `find_central_point` is just about returning the cube that matches the constraints passed in.  `self.central_point` is provided in units of `self.parameter_units` (see class `__init__`, so they will be consistent.  This function finds the value of the coordinate `central_point` in CUBE units, then returns the slice of the cube that matches this value.

I don't know what happens if `parameter_units` is None, I doubt there is a use case for this (since the value we are extracting should always be numerical and therefore have at least a unit of 1).  I haven't changed functionality here except to fix a bug.

Re unit tests: all I've done here is to replace cube setup so that time coordinates have units of "seconds", and so that this is controlled centrally.  Work to change plugin functionality and test coverage is probably out of scope, but could be opened as follow on work / technical debt if you think it's necessary?

Reply:
self.parameter_units is not None so parameter_units will be set to self.parameter_units. However the central_point has been convert to the units of the cube so the message could be inconsistent (e.g. if self.central_point=1 self.parameter_units='hour' but the cube is in seconds the message would be "The central point 3600 in units of hours is not available".  This is a bug that needs to be fixed, you haven't introduced although it would have been correct if you had left it as self.central_point.

Reply:
Ah OK.  Good spot.  Will fix that.

Reply:
Also I now understand your initial comment - you are right, the test on `parameter_units` makes no sense at all.  I'll remove that (and check it doesn't break anything before pushing the change).